### DISCUSSION TITLE
Create SSH Certificate

### DISCUSSION THREAD

**Discussion by CoderMcDaniel** [smallstep/certificates#1935] [Q&A]
I am using the 0.26.2 step-ca container.  I have initialized the CA with the following: `step ca init --ssh --root /home/step/certs/root_ca.crt --key /home/step/secrets/root_ca_key`

I am then trying to create a user SSH certificate with the following: `step ssh certificate <user> id_ecdsa`.  When I run this command, I select the SSHPOP provisioner created at initialization.  This runs and gives me the following message:  "SSHPOP requires the '--sshpop-cert' flag".

I cannot figure out how to implement that flag.   Any ideas?

Thanks!

**Comment by tashian** [comment]
Hi!

The `SSHPOP` provisioner is only for renewals of SSH host certificates.
It enables renewal automation on SSH hosts.
You should be able to get a certificate using the JWK provisioner that was created with your CA.
But, you'll probably want to reserve that provisioner for admin purposes and create other provisioners for getting SSH user and host certificates. For example, the OpenID Connect provisioner is popular for user certificates because it enables [SSO for SSH](https://smallstep.com/blog/diy-single-sign-on-for-ssh/).

Hope this helps—

**Reply by CoderMcDaniel** [reply]
This helps.  Appreciate it.  The `step ssh` documentation [here](https://smallstep.com/docs/step-cli/reference/ssh/#name) is a little confusing. It states to create the key pair and certificate use the `step ssh certificate` command.  But perhaps it's correct and I should be using a different provisioner than the SSHPOP as you stated.  Will report back after I give it a try.

Thanks again!

**Reply by CoderMcDaniel** [reply]
@tashian  When I select the JWK provisioner, running the step ssh certificate command, I get "The requested method is not implemented by the certificate authority."  When I deployed the container to run step-ca, I utilized the `DOCKER_STEPCA_INIT_SSH=1` environment variable. Is there anything else required to setup the SSH keys and certificate creation?

Thanks!

**Reply by cocoon** [reply]
there was a wrong wording for this env var, it needed to be DOCKER_STEPCA_INIT_SSH=true
https://github.com/smallstep/docs/issues/419

to add it manually to an already running ca:
https://github.com/smallstep/certificates/discussions/400
